ouroboros-consensus/test/consensus-test/Test/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -237,18 +301,42 @@ data ChainSyncOutcome = ChainSyncOutcome {
 -- Note that updates that are scheduled before the time at which we start
 -- syncing help generate different chains to start syncing from.
 runChainSync
-    :: forall m. IOLike m
-    => SecurityParam
+    :: forall m. (IOLike m, MonadTime m)
+    => ClockSkew
+    -> SecurityParam
     -> ClientUpdates
     -> ServerUpdates
     -> InvalidBlocks
     -> Tick  -- ^ Start chain syncing at this time
     -> m ChainSyncOutcome
-runChainSync securityParam (ClientUpdates clientUpdates)
+runChainSync skew securityParam (ClientUpdates clientUpdates)
     (ServerUpdates serverUpdates) (InvalidBlocks invalidBlocks)
     startSyncingAt = withRegistry $ \registry -> do
 
-    clock <- LogicalClock.new registry numTicks
+    systemTime <- do
```

Comment:
> `clientSystemTime :: SystemTime m <-`

I generally avoid pattern guards with type variables, since they have subtle/sometimes surprising rules for binding variables. That's my only reason.

Good rename 👍